pkg/kubectl/cmd/create_pvc_test.go

```diff
@@ -0,0 +1,328 @@
+				Name:        "mypvc",
+				AccessModes: []api.PersistentVolumeAccessMode{api.ReadOnlyMany},
+			},
+			expectErr: true,
+		},
+		"test-valid-case": {
+			pvcOptions: &CreatePVCOptions{
+				Name:        "mypvc",
+				AccessModes: []api.PersistentVolumeAccessMode{api.ReadOnlyMany},
+				Capacity:    &capacityQuantity,
+			},
+			expectErr: false,
+		},
+	}
+
+	for name, test := range tests {
+		err := test.pvcOptions.Validate()
+		if test.expectErr && err == nil {
+			t.Errorf("%s: expect error happens but validate passes.", name)
+		}
+		if !test.expectErr && err != nil {
+			t.Errorf("%s: unexpected error: %v", name, err)
+		}
+	}
+}
+
+func TestCreatePVCOptionsComplete(t *testing.T) {
+	pvcName, emptyStorageClass, storageClass := "mypvc", "", "foo"
+	capacityQuantity, _ := resource.ParseQuantity("1Gi")
+
+	f, tf, _, _ := cmdtesting.NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.Namespace = "test"
+	tf.Client = &fake.RESTClient{}
+	tf.ClientConfig = defaultClientConfig()
+
+	buf := bytes.NewBuffer([]byte{})
+
+	tests := map[string]struct {
+		params                []string
+		accessMode            string
+		capacity              string
+		storageClass          string
+		storageClassSpecified bool
+		expected              *CreatePVCOptions
+		expectErr             bool
+	}{
+		"test-missing-name": {
+			params:     []string{},
+			accessMode: "RWO",
+			capacity:   "1Gi",
+			expectErr:  true,
+		},
+		"test-accessmode-abbreviation": {
+			params:     []string{pvcName},
+			accessMode: "RWO",
+			capacity:   "1Gi",
+			expected: &CreatePVCOptions{
+				Name: pvcName,
+				AccessModes: []api.PersistentVolumeAccessMode{
+					api.ReadWriteOnce,
+				},
+				Capacity: &capacityQuantity,
+			},
+			expectErr: false,
+		},
+		"test-invalid-capacity": {
+			params:     []string{pvcName},
+			accessMode: "RWO",
+			capacity:   "invalid",
+			expectErr:  true,
+		},
+		"test-empty-storageclass": {
+			params:                []string{pvcName},
+			accessMode:            "RWO",
+			capacity:              "1Gi",
+			storageClassSpecified: true,
+			expected: &CreatePVCOptions{
+				Name: pvcName,
+				AccessModes: []api.PersistentVolumeAccessMode{
+					api.ReadWriteOnce,
+				},
+				Capacity:     &capacityQuantity,
+				StorageClass: &emptyStorageClass,
+			},
+			expectErr: false,
+		},
+		"test-valid-complete-case": {
+			params:                []string{pvcName},
+			accessMode:            "ReadWriteOnce",
+			capacity:              "1Gi",
+			storageClass:          "foo",
+			storageClassSpecified: true,
+			expected: &CreatePVCOptions{
+				Name: pvcName,
+				AccessModes: []api.PersistentVolumeAccessMode{
+					api.ReadWriteOnce,
+				},
+				Capacity:     &capacityQuantity,
+				StorageClass: &storageClass,
+			},
+			expectErr: false,
+		},
+	}
+
+	for name, test := range tests {
+		pvcOptions := &CreatePVCOptions{}
+		cmd := NewCmdCreatePersistentVolumeClaim(f, buf)
+
+		cmd.Flags().Set("access-mode", test.accessMode)
```

Comment:
Please add a test with more that one access mode.

Reply:
ok, let me add one